src/Microsoft.AspNetCore.Http/Internal/DefaultHttpRequest.cs

```diff
@@ -41,7 +41,7 @@ public virtual void Uninitialize()
             _features.Fetch(ref _features.Cache.Query, f => new QueryFeature(f));
 
         private IFormFeature FormFeature =>
-            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f));
+            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f, (f._context as DefaultHttpContext)?.FormOptions));
```

Comment:
Pushing it into the init means the .ctor needs to change to
```csharp
public DefaultHttpContext(IFeatureCollection features, ObjectPool<StringBuilder> builderPool, FormOptions formOptions)
{
    Initialize(features);

    var defaultRequest = (_request as DefaultHttpRequest);
    if (defaultRequest != null)
    {
        defaultRequest.FormOptions = formOptions;
    }

    var defaultResponse = (_response as DefaultHttpResponse);
    if (defaultResponse != null)
    {
        defaultResponse.BuilderPool = builderPool;
    }
}
```
So you are doing to `as` tests upfront whether you use them or not?